db.go

```diff
@@ -230,7 +230,7 @@ func (db *DB) getValueByRecord(record *Record) ([]byte, error) {
 	}
 
 	// firstly we find data in cache
-	if db.hintKeyAndRAMIdxModeLru.Get(record) != nil {
+	if db.getHintKeyAndRAMIdxCacheSize() > 0 && db.hintKeyAndRAMIdxModeLru.Get(record) != nil {
```

Comment:
Can we just do `db.hintKeyAndRAMIdxModeLru.Get(record)` and store it as a variable and avoid to visit LRU twice.